astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
+    PyUnitListProxy *b,
+    int op) {
+
+  int equal;
+  int status;
+  struct PyUnitListProxy *ax;
+  struct PyUnitListProxy *bx;
+   
+  
+  if ((op == Py_EQ || op == Py_NE) &&
+      PyObject_TypeCheck(b, &PyUnitListProxyType)) {
+      ax = &((PyUnitListProxy *)a)->pyobject;
+      bx = &((PyUnitListProxy *)b)->pyobject;
+          
+    
+    if (!PyObject_RichCompareBool(a->unit_class, b->unit_class,Py_EQ)||
+        a->size != b->size||
+        strncmp(a->array, b->array, 72) ) {
+        status = 0;
+        equal = 0;
+       }
+
+    else if (ax == NULL || bx == NULL){
+       status = 1;
+      }
+
+    else {
+       status = 0;
+       equal = 1;
+      }
+
+    if (status == 0) {
+      if (op == Py_NE) {
+        equal = !equal;
+       }
+      if (equal) {
+        Py_RETURN_TRUE;
+      } 
+      else {
+        Py_RETURN_FALSE;
+       }
+      } 
+  }
+
+  Py_INCREF(Py_NotImplemented);
```

Comment:
There is [`Py_RETURN_NOTIMPLEMENTED`](https://docs.python.org/3/c-api/object.html#c.Py_RETURN_NOTIMPLEMENTED) at least in Python > 3.5 (which is all we currently support).